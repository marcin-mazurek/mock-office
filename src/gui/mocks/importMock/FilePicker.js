import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { initAction } from './actions';
import uploadIcon from '../../assets/icons_gray_upload@3x.svg';

export class FilePicker extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    if (e.target.files.length) {
      this.props.init(this.props.server, this.props.scenario, e.target.files);
    }
  }

  render() {
    const { handleChange } = this;

    return (
      <div className="file-picker">
        <label
          className="file-picker__button inspect-server-mocks-header__button"
          htmlFor="file-picker__input"
        >
          <img src={uploadIcon} role="presentation" style={{ marginRight: '11px' }} />
          Import from file
        </label>
        <input
          id="file-picker__input"
          className="file-picker__input"
          type="file"
          onChange={handleChange}
          value=""
        />
      </div>
    );
  }
}
FilePicker.propTypes = {
  server: PropTypes.string.isRequired,
  init: PropTypes.func.isRequired,
  scenario: PropTypes.string.isRequired
};

const mapDispatchToProps = {
  init: initAction
};
export default connect(null, mapDispatchToProps)(FilePicker);
