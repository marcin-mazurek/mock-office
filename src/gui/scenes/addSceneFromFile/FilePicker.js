import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { init as intiAddSceneFromFile } from './actions';
import uploadIcon from '../../assets/icons_gray_upload@3x.svg';

const mapDispatchToProps = {
  intiAddSceneFromFile
};

export class FilePicker extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    if (e.target.files.length) {
      this.props.intiAddSceneFromFile(this.props.serverId, e.target.files);
    }
  }

  render() {
    const { handleChange } = this;

    return (
      <div className="file-picker">
        <label
          className="file-picker__button inspect-server-scenes-header__button"
          htmlFor="file-picker__input"
        >
          <img src={uploadIcon} role="presentation" style={{ marginRight: '11px' }} />
          Import from file
        </label>
        <input
          id="file-picker__input"
          className="file-picker__input"
          type="file"
          onChange={handleChange}
          value=""
        />
      </div>
    );
  }
}

FilePicker.propTypes = {
  serverId: PropTypes.string.isRequired,
  intiAddSceneFromFile: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(FilePicker);
