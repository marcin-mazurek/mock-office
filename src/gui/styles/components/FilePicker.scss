.file-picker {
  position: relative;
  overflow: hidden;
}

.file-picker__button {
  @extend %button-reset;
  cursor: pointer;
  padding: 0 20px;
}

.file-picker__input {
  display: none;
}
