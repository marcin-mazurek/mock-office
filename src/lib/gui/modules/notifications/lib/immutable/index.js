export * from './selectors';
export { default as NotificationsConnect } from './NotificationsConnect';
