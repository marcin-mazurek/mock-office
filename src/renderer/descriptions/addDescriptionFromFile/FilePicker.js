import React from 'react';
import { connect } from 'react-redux';
import { init as intiAddDescriptionFromFile } from './actions';

const mapDispatchToProps = {
  intiAddDescriptionFromFile
};

export class FilePicker extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    if (e.target.files.length) {
      this.props.intiAddDescriptionFromFile(this.props.serverId, e.target.files);
    }
  }

  render() {
    const { handleChange } = this;

    return (
      <div className="file-picker">
        <label
          className="file-picker__button button inspect-server-descriptions-header__button"
          htmlFor="file-picker__input"
        >
          <i className="fa fa-upload" />
        </label>
        <input
          id="file-picker__input"
          className="file-picker__input"
          type="file"
          onChange={handleChange}
          value=""
        />
      </div>
    );
  }
}

FilePicker.propTypes = {
  serverId: React.PropTypes.string.isRequired,
  intiAddDescriptionFromFile: React.PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(FilePicker);
