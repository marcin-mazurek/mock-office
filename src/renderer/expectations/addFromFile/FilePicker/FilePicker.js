import React from 'react';
import { connect } from 'react-redux';
import { filePick } from '../../actions';

const mapDispatchToProps = {
  filePick
};

class FilePicker extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.props.filePick(this.fileNode.files);
  }

  render() {
    const { handleChange } = this;

    return (
      <div>
        <input type="file" ref={(node) => { this.fileNode = node; }} onChange={handleChange} />
      </div>
    );
  }
}

FilePicker.propTypes = {
  filePick: React.PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(FilePicker);
