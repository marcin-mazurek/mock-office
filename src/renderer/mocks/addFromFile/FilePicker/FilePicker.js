import React from 'react';
import { connect } from 'react-redux';

const mapDispatchToProps = {
  onChange: files => ({ type: 'mocks/FILE_PICKED', files })
};

class FilePicker extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.props.onChange(this.fileNode.files);
  }

  render() {
    const { handleChange } = this;

    return (
      <div>
        <input type="file" ref={(node) => { this.fileNode = node; }} onChange={handleChange} />
      </div>
    );
  }
}

FilePicker.propTypes = {
  onChange: React.PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(FilePicker);
