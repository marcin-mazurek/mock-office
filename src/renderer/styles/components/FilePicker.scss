.file-picker {
  position: relative;
  overflow: hidden;
  cursor: pointer;
}

.file-picker__button {
  border: $file-picker-button-border;
}

.file-picker__input {
  position: absolute;
  top: 0;
  right: 0;
  margin: 0;
  padding: 0;
  cursor: pointer;
  opacity: 0;
  width: 100%;
  height: 100%;

  button {
    cursor: pointer;
  }
}
