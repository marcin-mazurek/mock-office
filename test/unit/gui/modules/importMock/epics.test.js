import { Map } from 'immutable';
import configureMockStore from 'redux-mock-store';
import 'rxjs';
import { createEpicMiddleware } from 'redux-observable';
import { filePickedAction } from '../../../../../src/gui/components/FilePicker';

describe('importMockEpic', () => {
  let store;

  afterEach(() => {
    store.clearActions();
  });

  test('snapshot of success flow', (done) => {
    jest.mock('../../../../../src/gui/resources/api', () => ({
      addMock() {
        return Promise.resolve({
          data: {
            id: 'mock-id',
            tasks: []
          }
        });
      }
    }));
    const addMockFromFileEpic = require('../../../../../src/gui/epics/importMock').default;
    const epicMiddleware = createEpicMiddleware(addMockFromFileEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    store = mockStore(new Map({
      mocks: new Map({
        entities: new Map({
          'mock-id': {
            type: 'server type',
            name: 'server name',
            port: 3000,
            id: 'mock-id'
          }
        })
      })
    }));

    store.dispatch(filePickedAction('server-id', 'scenario-id', [
      new Blob(['[{"tasks":[]}]'])
    ]));
    setTimeout(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    }, 100);
  });
});
