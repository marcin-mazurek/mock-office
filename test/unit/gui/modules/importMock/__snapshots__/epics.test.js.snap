// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importMockEpic snapshot of success flow 1`] = `
Array [
  Object {
    "files": Array [
      Blob {
        Symbol(impl): BlobImpl {
          "_buffer": Object {
            "data": Array [
              91,
              123,
              34,
              116,
              97,
              115,
              107,
              115,
              34,
              58,
              91,
              93,
              125,
              93,
            ],
            "type": "Buffer",
          },
          "isClosed": false,
          "type": "",
          Symbol(wrapper): [Circular],
        },
      },
    ],
    "scenario": "scenario-id",
    "server": "server-id",
    "type": "component/FilePicker/FILE_PICKED",
  },
  Object {
    "mocks": Array [
      undefined,
    ],
    "scenario": undefined,
    "type": "importMock/SUCCEEDED",
  },
]
`;
